pkg/controllers/taint/clustertaintpolicy_controller.go

```diff
@@ -217,30 +232,35 @@ func (c *ClusterTaintPolicyController) SetupWithManager(mgr controllerruntime.Ma
 		DeleteFunc: func(_ event.DeleteEvent) bool { return false },
 	}
 
-	clusterTaintPolicyMapFunc := handler.MapFunc(
-		func(ctx context.Context, policyObj client.Object) []reconcile.Request {
-			clusterList := &clusterv1alpha1.ClusterList{}
+	clusterMapFunc := handler.MapFunc(
+		func(ctx context.Context, clusterObj client.Object) []reconcile.Request {
+			clusterTaintPolicyList := &policyv1alpha1.ClusterTaintPolicyList{}
 			listOption := &client.ListOptions{UnsafeDisableDeepCopy: ptr.To(true)}
-			if err := c.Client.List(ctx, clusterList, listOption); err != nil {
-				klog.Errorf("Failed to list Cluster, error: %v", err)
+			if err := c.Client.List(ctx, clusterTaintPolicyList, listOption); err != nil {
+				klog.Errorf("Failed to list ClusterTaintPolicy, error: %v", err)
 				return nil
 			}
 
-			policy := policyObj.(*policyv1alpha1.ClusterTaintPolicy)
+			policies := clusterTaintPolicyList.Items
+			sort.Slice(policies, func(i, j int) bool {
+				return policies[i].CreationTimestamp.Before(&policies[j].CreationTimestamp)
+			})
```

Comment:
Sorting the policy here won't help as these policies are not proceed as this order, especially in concurrent cases.

Reply:
In this way, when we use ClusterTaintPolicy as the reconcile object, maybe we cannot guarantee the idempotent of the execution result.